library/std/src/sync/mutex.rs

```diff
@@ -480,13 +480,7 @@ impl<T: ?Sized + fmt::Debug> fmt::Debug for Mutex<T> {
                 d.field("data", &&**err.get_ref());
             }
             Err(TryLockError::WouldBlock) => {
-                struct LockedPlaceholder;
-                impl fmt::Debug for LockedPlaceholder {
-                    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                        f.write_str("<locked>")
-                    }
-                }
-                d.field("data", &LockedPlaceholder);
+                d.field("data", &fmt::DebugPlaceholder("<locked>"));
```

Comment:
I think we can achieve equivalent results in all of these cases with `format_args!("<locked>")` as the thing passed in, which avoids the extra type and works on stable.

Reply:
Oh, yeah, that works!

Reply:
I've changed it now.